src/rich-text-editor.component.js

```diff
@@ -95,9 +95,15 @@ export const RichTextEditor = forwardRef((props, editorRef) => {
   }, [isFocused()])
 
   useEffect(() => {
-    richTextContext.selectRangeFromBeforeBlur = () => {
+    richTextContext.selectRangeFromBeforeBlur = (options = {usePreviousRange: false}) => {
       if (divRef.current && document.activeElement !== divRef.current && !divRef.current.contains(document.activeElement)) {
-        divRef.current.focus()
+        if (options.usePreviousRange && selectionRangeBeforeBlurRef.current) {
+          const selection = window.getSelection()
+          selection.removeAllRanges()
+          selection.addRange(selectionRangeBeforeBlurRef.current)
```

Comment:
With bolding/italic/underline, calling focus() is the right thing to do. But with links and text-as-image, you need to do the previous range thing.

Why?? honestly idk. It might have something to do with how links often have a more complicated UX where there are other inputs that take the focus so that by the time that you refocus the contentEditable div, it has forgotten the previous selection.

Reply:
See https://canopytax.atlassian.net/browse/BLUE-665